src/components/DocumentView/BlockSyncedBlock.tsx

```diff
@@ -13,22 +14,34 @@ export async function BlockSyncedBlock(props: BlockProps<DocumentBlockSyncedBloc
         return null;
     }
 
-    const result = await getSyncedBlock(
+    const syncedBlock = await getSyncedBlock(
         apiToken,
         block.data.ref.organization,
         block.data.ref.syncedBlock,
     );
 
-    if (!result) {
+    if (!syncedBlock) {
         return null;
     }
 
     return (
         <Blocks
-            nodes={result.document.nodes}
-            document={result.document}
+            nodes={syncedBlock.document.nodes}
+            document={syncedBlock.document}
             ancestorBlocks={[...ancestorBlocks, block]}
-            context={context}
+            context={{
+                ...context,
+                resolveContentRef: async (ref, options) => {
+                    if (!syncedBlock?.files) {
+                        return context.resolveContentRef(ref, options);
```

Comment:
This is incorrect and will not match the behaviour in the editor.

If I copy a link `{ kind: "page", page: 'something' }` in a sync block, it'll do a broken link in the app, but will work in published content if the page `something` exists in the space.

Resolution of content ref in the synced block should be isolated from the one for the rest of the content (clean context). Except if you tell me the logic has changed in the app.

Reply:
This is the way it should work. If a customer use a link in a synced block and use its synced block in only one space it's better.